packages/graphql/tests/integration/directives/populatedBy.int.test.ts

```diff
@@ -1465,6 +1465,972 @@ describe("@populatedBy directive", () => {
+            });
+        });
+
+        describe("@populatedBy - Date", () => {
+            test("Should use on CREATE", async () => {
+                const testMovie = testHelper.createUniqueType("Movie");
+
+                const date = new Date();
+
+                const callback = () => Promise.resolve(date.toISOString());
+
+                const typeDefs = gql`
+                    type ${testMovie.name} {
+                        id: ID
+                        callback: Date! @populatedBy(operations: [CREATE], callback: "callback")
+                    }
+                `;
+
+                await testHelper.initNeo4jGraphQL({
+                    typeDefs,
+                    features: {
+                        populatedBy: {
+                            callbacks: {
+                                callback,
+                            },
+                        },
+                    },
+                });
+
+                const movieId = generate({
+                    charset: "alphabetic",
+                });
+
+                const mutation = `
+                    mutation {
+                        ${testMovie.operations.create}(input: [{ id: "${movieId}" }]) {
+                            ${testMovie.plural} {
+                                id
+                                callback
+                            }
+                        }
+                    }
+                `;
+
+                const result = await testHelper.executeGraphQL(mutation);
+
+                expect(result.errors).toBeUndefined();
+                expect(result.data as any).toMatchObject({
+                    [testMovie.operations.create]: {
+                        [testMovie.plural]: [
+                            {
+                                id: movieId,
+                                callback: date.toISOString().split("T")[0],
+                            },
+                        ],
+                    },
+                });
+            });
+
+            test("Should use on UPDATE", async () => {
+                const testMovie = testHelper.createUniqueType("Movie");
+
+                const date = new Date();
+
+                const callback = () => Promise.resolve(date.toISOString());
+
+                const typeDefs = gql`
+                    type ${testMovie.name} {
+                        id: ID
+                        callback: Date! @populatedBy(operations: [UPDATE], callback: "callback")
+                    }
+                `;
+
+                await testHelper.initNeo4jGraphQL({
+                    typeDefs,
+                    features: {
+                        populatedBy: {
+                            callbacks: {
+                                callback,
+                            },
+                        },
+                    },
+                });
+
+                const movieId = generate({
+                    charset: "alphabetic",
+                });
```

Comment:
Maybe we can avoid randomly generated test data? This makes debugging harder and may cause flaky tests

Reply:
I think this should probably be addressed in a holistic future change